jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorization.java

```diff
@@ -94,4 +166,16 @@ public boolean equals(Object other) {
   public int hashCode() {
     return Objects.hash(scheme, token);
   }
+
+  /**
+   * Check if this authorization allows accessing the specified repository.
+   *
+   * @param repository repository in question
+   * @param access the access scope ("push" or "pull")
+   * @return true if the repository was covered
+   */
+  public boolean canAccess(String repository, String access) {
+    // if null then we assume that all repositories are granted
```

Comment:
Worth explaining this in the Javadoc.

Reply:
Yeah I don't understand why "null" is valid

Reply:
The other non-Docker Container Registry Token methods don't provide this information.

Reply:
I think the method name and the Javadoc can still be confusing. I think we just need to clarify (and justify). For example, I can create an `Authorization` instance with `fromBasicCredentials("my-username", "my-passwd")`. Then `canAccess("any repository", "pull or push or anything")` will always return true. I feel this is not intuitive.

Reply:
What about making this specific to the blob-mount: `canAttemptBlobMount(repository)`

Reply:
That technically works. Man, naming is hard. I see depending on the name, we may want to relocate this class from the `.http` package. (Relocation can be done later.) This class is meant for holding the very string value for the Authorization HTTP header (i.e., for `Authorization: <some schema> <token value>`) as explained in the class Javadoc, hence was holding the two fields `schema` and `token`. I thought `repositoryGrants` was OK, because it is actually information in `token`. But if we incorporate the notions of BLOb or cross-repo mount, it may make sense to get this out of `.http`.

That said, I don't have a strong opinion about what the name should be. Maybe the logic for `canAttemptBlobMount` can be outside of this method, I don't know.

Reply:
Hmm good point.  I'll move this into `RegistryClient` — it's only called from `pushBlobStep`.